### PR TITLE
fix: Mint tokens view - exclude watch accounts from accounts dropdown

### PR DESCRIPTION
**Pull request by alexjba**
### What does the PR do
Closing: #11162 

Exclude watch accounts from mint token-accounts dropdown by adding filter for watch accounts in the accounts model.

### Affected areas
Mint tokens - account selection

https://github.com/status-im/status-desktop/assets/47811206/d6d79cd0-f095-4e2a-9dcd-ee65c29e675e

### CODE REVIEW

**Comment by alexjba** [comment]
>Now the first account is not selected, seems to be the last one, by default.

@noeliaSD I could not reproduce this part. Please let me know if it's still reproducible on your account.

File: ui/app/AppLayouts/Chat/views/communities/CommunityNewTokenView.qml [hunk 1]
```diff
@@ -225,6 +225,11 @@ StatusScrollView {
                         expression: { return getColor(model.colorId) }
                     }
                 ]
+                filters: ValueFilter {
+                        roleName: "walletType"
+                        value: Constants.watchWalletType
+                        inverted: true
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/views/communities/CommunityNewTokenView.qml:231]: What about keycard? (aka `Constants.keyWalletType`), thouse should imo be excluded too

Comment (alexjba) [reply ui/app/AppLayouts/Chat/views/communities/CommunityNewTokenView.qml:231]: Why? IDK if there's any difference between a wallet account imported from seed phrase or keycard from this perspective. Both wallets should be owned by the user 🤔

Comment (alexjba) [reply ui/app/AppLayouts/Chat/views/communities/CommunityNewTokenView.qml:231]: @noeliaSD Any thoughts on this?

Comment (noeliaSD) [reply ui/app/AppLayouts/Chat/views/communities/CommunityNewTokenView.qml:231]: I don't have the full information here but my understanding is.. Here we should display all wallet addresses where I can do transactions. 

@benjthayer could you confirm?

Comment (benjthayer) [reply ui/app/AppLayouts/Chat/views/communities/CommunityNewTokenView.qml:231]: > I don't have the full information here but my understanding is.. Here we should display all wallet addresses where I can do transactions.
> 
> @benjthayer could you confirm?

That would be my understanding - if the user has the ability to transact using the account (i.e. not watch only) the account should be able to be used for minting purposes. A Keycard account would just require the user to authenticate the transaction using the relevant Keycard at this point:
https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?type=design&node-id=29384-568047&t=1KMcB8L5M6fBcY58-4

**Review by noeliaSD — COMMENTED** [review]
Code LGTM. I've not tested yet but watching the video I see the first account doesn't have `name`. I see in the wallet view it's indeed not shown so.. I guess if we can have an account with empty name in wallet.. Should we add some "default" text in this combo since it seems there is something wrong with an empty combo entry? WDYT?

**Review by caybro — APPROVED** [review]
LGTM

**Comment by noeliaSD** [comment]
> > Code LGTM. I've not tested yet but watching the video I see the first account doesn't have `name`. I see in the wallet view it's indeed not shown so.. I guess if we can have an account with empty name in wallet.. Should we add some "default" text in this combo since it seems there is something wrong with an empty combo entry? WDYT?
> 
> IMO the correct fix would be to validate the name in wallet. Might create confusion if we use different wallet names across the app. WDYT on creating a different issue for this?

Ok, yes, we can create a new issue for this with lower priority! Could you do it? Thanks

**Review by noeliaSD — APPROVED** [review]

**Pull request merged**